commerce-service/src/main/java/com/liferay/commerce/internal/messaging/CommerceOrderStatusMessageListener.java

```diff
@@ -44,17 +53,46 @@
 	protected void doReceive(Message message) throws Exception {
 		int orderStatus = message.getInteger("orderStatus");
 
-		if (orderStatus == CommerceOrderConstants.ORDER_STATUS_TO_TRANSMIT) {
-			long commerceOrderId = message.getLong("commerceOrderId");
+		long commerceOrderId = message.getLong("commerceOrderId");
 
-			CommerceOrder commerceOrder =
-				_commerceOrderLocalService.getCommerceOrder(commerceOrderId);
+		CommerceOrder commerceOrder =
+			_commerceOrderLocalService.getCommerceOrder(commerceOrderId);
+
+		CommerceChannel commerceChannel =
+			_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
+				commerceOrder.getGroupId());
+
+		boolean hasWorkflow =
+			_workflowDefinitionLinkLocalService.hasWorkflowDefinitionLink(
+				commerceOrder.getCompanyId(), commerceOrder.getGroupId(),
+				CommerceOrder.class.getName(), 0,
+				CommerceOrderConstants.TYPE_PK_APPROVAL);
+
+		if (hasWorkflow &&
+			(orderStatus == CommerceOrderConstants.ORDER_STATUS_OPEN) &&
+			(commerceOrder.getStatus() != WorkflowConstants.STATUS_APPROVED)) {
 
 			// Commerce notifications
 
-			CommerceChannel commerceChannel =
-				_commerceChannelLocalService.getCommerceChannelByOrderGroupId(
-					commerceOrder.getGroupId());
+			_commerceNotificationHelper.sendNotifications(
+				commerceChannel.getSiteGroupId(),
+				OrderSubmittedCommerceNotificationTypeImpl.KEY, commerceOrder);
+		}
+		else if (hasWorkflow &&
+				 (orderStatus == CommerceOrderConstants.ORDER_STATUS_OPEN) &&
+				 (commerceOrder.getStatus() ==
+					 WorkflowConstants.STATUS_APPROVED)) {
+
+			// Commerce notifications
```

Comment:
Let's remove these `// Commerce notifications` comments -- I think they just add noise at this point.